src/test/java/org/opentripplanner/middleware/utils/ItineraryUtilsTest.java

```diff
@@ -0,0 +1,377 @@
+
+        List<Itinerary> itineraries = otpDispatcherPlanResponse.getResponse().plan.itineraries;
+
+        ItineraryUtils.updateTripWithVerifiedItinerary(trip, itineraries);
+
+        Assertions.assertEquals(itineraries.get(1), trip.itinerary);
+    }
+
+    @ParameterizedTest
+    @MethodSource("createSameDayTestCases")
+    void testIsSameDay(SameDayTestCase testCase) throws URISyntaxException {
+        MonitoredTrip trip = makeTestTrip(testCase.isArrival);
+        // The time zone for trip (and testCase.tripTime) is US Eastern per trip location.
+        ZoneId zoneId = trip.timezoneForTargetLocation();
+
+        Itinerary itinerary = simpleItinerary(testCase.tripTime, testCase.isArrival);
+        Assertions.assertEquals(
+            testCase.shouldBeSameDay,
+            ItineraryUtils.isSameDay(itinerary, QUERY_DATE, testCase.timeOfDay, zoneId, testCase.isArrival),
+            testCase.getMessage(zoneId)
+        );
+    }
+
+    private static List<SameDayTestCase> createSameDayTestCases() {
+        return List.of(
+            // Same-day departures
+            new SameDayTestCase(QUERY_TIME, _2020_08_13__03_00_00_EDT, false, true),
+            new SameDayTestCase(QUERY_TIME, _2020_08_13__23_59_59_EDT, false, true),
+            new SameDayTestCase(QUERY_TIME, _2020_08_14__02_59_59_EDT, false, true),
+            new SameDayTestCase("1:23", _2020_08_12__03_00_00_EDT, false, true),
+            new SameDayTestCase("1:23", _2020_08_12__23_59_59_EDT, false, true),
+            new SameDayTestCase("1:23", _2020_08_13__02_59_59_EDT, false, true),
+
+            // Not same-day departures
+            new SameDayTestCase(QUERY_TIME, _2020_08_12__23_59_59_EDT, false, false),
+            new SameDayTestCase(QUERY_TIME, _2020_08_13__02_59_59_EDT, false, false),
+            new SameDayTestCase(QUERY_TIME, _2020_08_14__03_00_00_EDT, false, false),
+            new SameDayTestCase("1:23", _2020_08_13__03_00_00_EDT, false, false),
+            new SameDayTestCase("1:23", _2020_08_13__23_59_59_EDT, false, false),
+            new SameDayTestCase("1:23", _2020_08_14__02_59_59_EDT, false, false),
+
+            // Same-day arrivals
+            new SameDayTestCase(QUERY_TIME, _2020_08_13__03_00_00_EDT, true, true),
+            new SameDayTestCase(QUERY_TIME, _2020_08_13__23_59_59_EDT, true, true),
+            new SameDayTestCase(QUERY_TIME, _2020_08_14__02_59_59_EDT, true, true),
+            new SameDayTestCase("1:23", _2020_08_12__03_00_00_EDT, true, true),
+            new SameDayTestCase("1:23", _2020_08_12__23_59_59_EDT, true, true),
+            new SameDayTestCase("1:23", _2020_08_13__02_59_59_EDT, true, true),
+
+            // Not same-day arrivals
+            new SameDayTestCase(QUERY_TIME, _2020_08_12__23_59_59_EDT, true, false),
+            new SameDayTestCase(QUERY_TIME, _2020_08_13__02_59_59_EDT, true, false),
+            new SameDayTestCase(QUERY_TIME, _2020_08_14__03_00_00_EDT, true, false),
```

Comment:
Since OTP shouldn't ever return a "arrive by" itinerary that ends after the date + time request parameters, I don't believe such a case would occur. But if it did, I guess this is what I'd expect for being the same day though, so I guess it's fine to keep these test cases.

Reply:
This comment applies to the remaining 3 cases as well.